volatility3/framework/plugins/linux/tracing/ftrace.py

```diff
@@ -114,21 +114,24 @@ def extract_hash_table_filters(
             An iterable of ftrace_func_entry structs
         """
 
+        if hasattr(ftrace_ops, "func_hash"):
+            ftrace_hash = ftrace_ops.func_hash.filter_hash
+        else:
+            ftrace_hash = ftrace_ops.filter_hash
+
         try:
-            current_bucket_ptr = ftrace_ops.func_hash.filter_hash.buckets.first
+            current_bucket_ptr = ftrace_hash.buckets.first
         except exceptions.InvalidAddressException:
             vollog.log(
                 constants.LOGLEVEL_VV,
                 f"ftrace_func_entry list of ftrace_ops@{ftrace_ops.vol.offset:#x} is empty/invalid. Skipping it...",
             )
-            return []
+            return
```

Comment:
Because of the yield statements, I think it's ok if the type signature for this isn't `Optional` (although I'm not sure how the interpreter would know that)...?

Reply:
I overhauled the types here and did a version bump. We have other places in the framework that also have Iterable instead of Generator in the typing, but will leave that for later.

Reply:
Sure, no problem, just wondered whether it needed to be optional, or would just return an empty iterable/generator?